src/relation/schema.rs

```diff
@@ -98,10 +101,124 @@ impl Schema {
         self.fields.iter()
     }
 
+    fn update(self, column: &Identifier, datatype: DataType) -> Schema {
```

Comment:
Everywhere else in the code, this kind of method is called `with` or `with_xxx` why use this name that suggests self will be mutated?